app/models/episode.rb

```diff
@@ -378,6 +385,22 @@ def feeder_cdn_host
     ENV["FEEDER_CDN_HOST"]
   end
 
+  def segment_range
+    1..segment_count.to_i
+  end
+
+  def build_contents
+    segment_range.map do |p|
+      contents.find { |c| c.position == p } || contents.build(position: p)
```

Comment:
Used for the frontend ... return just the segment_range filled in with new/unsaved contents.

The other half is `destroy_out_of_range_contents` below - after the episode is saved, deletes content that we're no longer using.